features/dd-sdk-android-ndk/src/main/cpp/datadog-native-lib.cpp

```diff
@@ -29,7 +29,13 @@ static struct Context {
 
 
 static const char *LOG_TAG = "DatadogNdkCrashReporter";
+
+#ifndef NDEBUG
```

Comment:
I feel like this is a lot of changes just to support the test. Maybe it makes sense to leave the mutex as `static` and instead create an `NDEBUG` wrapped function just to lock it.